// codemirror-lang-sparql/dist/index.js
import { LRParser } from '@lezer/lr';
import { LRLanguage, foldNodeProp, foldInside, LanguageSupport } from '@codemirror/language';
import { styleTags, tags } from '@lezer/highlight';

// This file was generated by lezer-generator. You probably shouldn't edit it.
const
  BASE = 1,
  PREFIX = 2,
  SELECT = 3,
  CONSTRUCT = 4,
  DESCRIBE = 5,
  WHERE = 6,
  DISTINCT = 7,
  REDUCED = 8,
  STR = 9,
  LANG = 10,
  LANGMATCHES = 11,
  DATATYPE = 12,
  ASK = 13,
  BOUND = 14,
  IRI = 15,
  URI = 16,
  BNODE = 17,
  RAND = 18,
  ABS = 19,
  CEIL = 20,
  FLOOR = 21,
  ROUND = 22,
  CONCAT = 23,
  STRLEN = 24,
  UCASE = 25,
  LCASE = 26,
  ENCODE_FOR_URI = 27,
  CONTAINS = 28,
  STRSTARTS = 29,
  STRENDS = 30,
  STRBEFORE = 31,
  STRAFTER = 32,
  YEAR = 33,
  MONTH = 34,
  DAY = 35,
  HOURS = 36,
  MINUTES = 37,
  SECONDS = 38,
  TIMEZONE = 39,
  TZ = 40,
  NOW = 41,
  UUID = 42,
  STRUUID = 43,
  MD5 = 44,
  SHA1 = 45,
  SHA256 = 46,
  SHA384 = 47,
  SHA512 = 48,
  COALESCE = 49,
  IF = 50,
  STRLANG = 51,
  STRDT = 52,
  SAMETERM = 53,
  ISIRI = 54,
  ISURI = 55,
  ISBLANK = 56,
  ISLITERAL = 57,
  ISNUMERIC = 58,
  COUNT = 59,
  SUM = 60,
  MIN = 61,
  MAX = 62,
  AVG = 63,
  SAMPLE = 64,
  GROUP_CONCAT = 65,
  SEPARATOR = 66,
  SUBSTR = 67,
  REPLACE = 68,
  REGEX = 69,
  EXISTS = 70,
  NOT = 71,
  IN = 72,
  GROUP = 73,
  BY = 74,
  HAVING = 75,
  ORDER = 76,
  ASC = 77,
  DESC = 78,
  LIMIT = 79,
  OFFSET = 80,
  VALUES = 81,
  UNDEF = 82,
  UNION = 83,
  OPTIONAL = 84,
  MINUS = 85,
  GRAPH = 86,
  SERVICE = 87,
  SILENT = 88,
  FILTER = 89,
  BIND = 90,
  AS = 91,
  FROM = 92,
  NAMED = 93,
  TRUE = 94,
  FALSE = 95,
  LOAD = 96,
  INTO = 97,
  TO = 98,
  CLEAR = 99,
  DROP = 100,
  CREATE = 101,
  ADD = 102,
  MOVE = 103,
  COPY = 104,
  INSERT = 105,
  DATA = 106,
  DELETE = 107,
  WITH = 108,
  DEFAULT = 109,
  ALL = 110,
  USING = 111,
  Comment = 112,
  Unit = 113,
  Query = 114,
  Prologue = 115,
  BaseDecl = 116,
  IriRef = 117,
  PrefixDecl = 118,
  Pname_ns = 119,
  SelectQuery = 120,
  SelectClause = 121,
  Var = 122,
  Expression = 123,
  ConditionalOrExpression = 124,
  ConditionalAndExpression = 125,
  ValueLogical = 126,
  RelationalExpression = 127,
  NumericExpression = 128,
  AdditiveExpression = 129,
  MultiplicativeExpression = 130,
  UnaryExpression = 131,
  PrimaryExpression = 132,
  BrackettedExpression = 133,
  BuiltInCall = 134,
  Aggregate = 135,
  String = 137,
  Nil = 139,
  ExpressionList = 140,
  SubstringExpression = 141,
  StrReplaceExpression = 142,
  RegexExpression = 143,
  ExistsFunc = 144,
  GroupGraphPattern = 145,
  SubSelect = 146,
  WhereClause = 147,
  SolutionModifier = 148,
  GroupClause = 149,
  GroupCondition = 150,
  FunctionCall = 151,
  Iri = 152,
  PrefixedName = 153,
  Pname_ln = 154,
  ArgList = 155,
  HavingClause = 156,
  HavingCondition = 157,
  Constraint = 158,
  OrderClause = 159,
  OrderCondition = 160,
  LimitOffsetClauses = 161,
  LimitClause = 162,
  Integer = 163,
  OffsetClause = 164,
  ValuesClause = 165,
  DataBlock = 166,
  InlineDataOneVar = 167,
  DataBlockValue = 168,
  RDFLiteral = 169,
  Langtag = 170,
  NumericLiteral = 172,
  NumericLiteralUnsigned = 173,
  Decimal = 174,
  Double = 175,
  NumericLiteralPositive = 176,
  IntegerPositive = 177,
  DecimalPositive = 178,
  DoublePositive = 179,
  NumericLiteralNegative = 180,
  IntegerNegative = 181,
  DecimalNegative = 182,
  DoubleNegative = 183,
  BooleanLiteral = 184,
  InlineDataFull = 185,
  GroupGraphPatternSub = 186,
  TriplesBlock = 187,
  TriplesSameSubjectPath = 188,
  VarOrTerm = 189,
  GraphTerm = 190,
  BlankNode = 191,
  Blank_node_label = 192,
  Anon = 193,
  PropertyListPathNotEmpty = 194,
  VerbPath = 195,
  Path = 196,
  PathAlternative = 197,
  PathSequence = 198,
  PathEltOrInverse = 199,
  PathElt = 200,
  PathPrimary = 201,
  A_predicate = 202,
  Keyword = 203,
  PathNegatedPropertySet = 204,
  PathOneInPropertySet = 205,
  PathMod = 206,
  VerbSimple = 207,
  ObjectListPath = 208,
  ObjectPath = 209,
  GraphNodePath = 210,
  TriplesNodePath = 211,
  CollectionPath = 212,
  BlankNodePropertyListPath = 215,
  ObjectList = 216,
  Object$1 = 217,
  GraphNode = 218,
  TriplesNode = 219,
  Collection = 220,
  BlankNodePropertyList = 221,
  PropertyListNotEmpty = 222,
  Verb = 223,
  VarOrIri = 224,
  PropertyListPath = 225,
  GraphPatternNotTriples = 227,
  GroupOrUnionGraphPattern = 228,
  OptionalGraphPattern = 229,
  MinusGraphPattern = 230,
  GraphGraphPattern = 231,
  ServiceGraphPattern = 232,
  Filter = 233,
  Bind = 234,
  InlineData = 235,
  NotExistsFunc = 236,
  IriOrFunction = 237,
  DatasetClause = 238,
  DefaultGraphClause = 239,
  SourceSelector = 240,
  NamedGraphClause = 241,
  ConstructQuery = 242,
  ConstructTemplate = 243,
  ConstructTriples = 244,
  TriplesSameSubject = 245,
  PropertyList = 246,
  TriplesTemplate = 247,
  DescribeQuery = 248,
  AskQuery = 249,
  Update = 250,
  Update1 = 251,
  Load = 252,
  GraphRef = 253,
  Clear = 254,
  GraphRefAll = 255,
  Drop = 256,
  Add = 257,
  GraphOrDefault = 258,
  Move = 259,
  Copy = 260,
  Create = 261,
  InsertData = 262,
  QuadData = 263,
  Quads = 264,
  QuadsNotTriples = 265,
  DeleteData = 266,
  DeleteWhere = 267,
  QuadPattern = 268,
  Modify = 269,
  DeleteClause = 270,
  InsertClause = 271,
  UsingClause = 272;

var tokens = /*#__PURE__*/Object.freeze({
  __proto__: null,
  ABS: ABS,
  ADD: ADD,
  ALL: ALL,
  AS: AS,
  ASC: ASC,
  ASK: ASK,
  AVG: AVG,
  A_predicate: A_predicate,
  Add: Add,
  AdditiveExpression: AdditiveExpression,
  Aggregate: Aggregate,
  Anon: Anon,
  ArgList: ArgList,
  AskQuery: AskQuery,
  BASE: BASE,
  BIND: BIND,
  BNODE: BNODE,
  BOUND: BOUND,
  BY: BY,
  BaseDecl: BaseDecl,
  Bind: Bind,
  BlankNode: BlankNode,
  BlankNodePropertyList: BlankNodePropertyList,
  BlankNodePropertyListPath: BlankNodePropertyListPath,
  Blank_node_label: Blank_node_label,
  BooleanLiteral: BooleanLiteral,
  BrackettedExpression: BrackettedExpression,
  BuiltInCall: BuiltInCall,
  CEIL: CEIL,
  CLEAR: CLEAR,
  COALESCE: COALESCE,
  CONCAT: CONCAT,
  CONSTRUCT: CONSTRUCT,
  CONTAINS: CONTAINS,
  COPY: COPY,
  COUNT: COUNT,
  CREATE: CREATE,
  Clear: Clear,
  Collection: Collection,
  CollectionPath: CollectionPath,
  Comment: Comment,
  ConditionalAndExpression: ConditionalAndExpression,
  ConditionalOrExpression: ConditionalOrExpression,
  Constraint: Constraint,
  ConstructQuery: ConstructQuery,
  ConstructTemplate: ConstructTemplate,
  ConstructTriples: ConstructTriples,
  Copy: Copy,
  Create: Create,
  DATA: DATA,
  DATATYPE: DATATYPE,
  DAY: DAY,
  DEFAULT: DEFAULT,
  DELETE: DELETE,
  DESC: DESC,
  DESCRIBE: DESCRIBE,
  DISTINCT: DISTINCT,
  DROP: DROP,
  DataBlock: DataBlock,
  DataBlockValue: DataBlockValue,
  DatasetClause: DatasetClause,
  Decimal: Decimal,
  DecimalNegative: DecimalNegative,
  DecimalPositive: DecimalPositive,
  DefaultGraphClause: DefaultGraphClause,
  DeleteClause: DeleteClause,
  DeleteData: DeleteData,
  DeleteWhere: DeleteWhere,
  DescribeQuery: DescribeQuery,
  Double: Double,
  DoubleNegative: DoubleNegative,
  DoublePositive: DoublePositive,
  Drop: Drop,
  ENCODE_FOR_URI: ENCODE_FOR_URI,
  EXISTS: EXISTS,
  ExistsFunc: ExistsFunc,
  Expression: Expression,
  ExpressionList: ExpressionList,
  FALSE: FALSE,
  FILTER: FILTER,
  FLOOR: FLOOR,
  FROM: FROM,
  Filter: Filter,
  FunctionCall: FunctionCall,
  GRAPH: GRAPH,
  GROUP: GROUP,
  GROUP_CONCAT: GROUP_CONCAT,
  GraphGraphPattern: GraphGraphPattern,
  GraphNode: GraphNode,
  GraphNodePath: GraphNodePath,
  GraphOrDefault: GraphOrDefault,
  GraphPatternNotTriples: GraphPatternNotTriples,
  GraphRef: GraphRef,
  GraphRefAll: GraphRefAll,
  GraphTerm: GraphTerm,
  GroupClause: GroupClause,
  GroupCondition: GroupCondition,
  GroupGraphPattern: GroupGraphPattern,
  GroupGraphPatternSub: GroupGraphPatternSub,
  GroupOrUnionGraphPattern: GroupOrUnionGraphPattern,
  HAVING: HAVING,
  HOURS: HOURS,
  HavingClause: HavingClause,
  HavingCondition: HavingCondition,
  IF: IF,
  IN: IN,
  INSERT: INSERT,
  INTO: INTO,
  IRI: IRI,
  ISBLANK: ISBLANK,
  ISIRI: ISIRI,
  ISLITERAL: ISLITERAL,
  ISNUMERIC: ISNUMERIC,
  ISURI: ISURI,
  InlineData: InlineData,
  InlineDataFull: InlineDataFull,
  InlineDataOneVar: InlineDataOneVar,
  InsertClause: InsertClause,
  InsertData: InsertData,
  Integer: Integer,
  IntegerNegative: IntegerNegative,
  IntegerPositive: IntegerPositive,
  Iri: Iri,
  IriOrFunction: IriOrFunction,
  IriRef: IriRef,
  Keyword: Keyword,
  LANG: LANG,
  LANGMATCHES: LANGMATCHES,
  LCASE: LCASE,
  LIMIT: LIMIT,
  LOAD: LOAD,
  Langtag: Langtag,
  LimitClause: LimitClause,
  LimitOffsetClauses: LimitOffsetClauses,
  Load: Load,
  MAX: MAX,
  MD5: MD5,
  MIN: MIN,
  MINUS: MINUS,
  MINUTES: MINUTES,
  MONTH: MONTH,
  MOVE: MOVE,
  MinusGraphPattern: MinusGraphPattern,
  Modify: Modify,
  Move: Move,
  MultiplicativeExpression: MultiplicativeExpression,
  NAMED: NAMED,
  NOT: NOT,
  NOW: NOW,
  NamedGraphClause: NamedGraphClause,
  Nil: Nil,
  NotExistsFunc: NotExistsFunc,
  NumericExpression: NumericExpression,
  NumericLiteral: NumericLiteral,
  NumericLiteralNegative: NumericLiteralNegative,
  NumericLiteralPositive: NumericLiteralPositive,
  NumericLiteralUnsigned: NumericLiteralUnsigned,
  OFFSET: OFFSET,
  OPTIONAL: OPTIONAL,
  ORDER: ORDER,
  Object: Object$1,
  ObjectList: ObjectList,
  ObjectListPath: ObjectListPath,
  ObjectPath: ObjectPath,
  OffsetClause: OffsetClause,
  OptionalGraphPattern: OptionalGraphPattern,
  OrderClause: OrderClause,
  OrderCondition: OrderCondition,
  PREFIX: PREFIX,
  Path: Path,
  PathAlternative: PathAlternative,
  PathElt: PathElt,
  PathEltOrInverse: PathEltOrInverse,
  PathMod: PathMod,
  PathNegatedPropertySet: PathNegatedPropertySet,
  PathOneInPropertySet: PathOneInPropertySet,
  PathPrimary: PathPrimary,
  PathSequence: PathSequence,
  Pname_ln: Pname_ln,
  Pname_ns: Pname_ns,
  PrefixDecl: PrefixDecl,
  PrefixedName: PrefixedName,
  PrimaryExpression: PrimaryExpression,
  Prologue: Prologue,
  PropertyList: PropertyList,
  PropertyListNotEmpty: PropertyListNotEmpty,
  PropertyListPath: PropertyListPath,
  PropertyListPathNotEmpty: PropertyListPathNotEmpty,
  QuadData: QuadData,
  QuadPattern: QuadPattern,
  Quads: Quads,
  QuadsNotTriples: QuadsNotTriples,
  Query: Query,
  RAND: RAND,
  RDFLiteral: RDFLiteral,
  REDUCED: REDUCED,
  REGEX: REGEX,
  REPLACE: REPLACE,
  ROUND: ROUND,
  RegexExpression: RegexExpression,
  RelationalExpression: RelationalExpression,
  SAMETERM: SAMETERM,
  SAMPLE: SAMPLE,
  SECONDS: SECONDS,
  SELECT: SELECT,
  SEPARATOR: SEPARATOR,
  SERVICE: SERVICE,
  SHA1: SHA1,
  SHA256: SHA256,
  SHA384: SHA384,
  SHA512: SHA512,
  SILENT: SILENT,
  STR: STR,
  STRAFTER: STRAFTER,
  STRBEFORE: STRBEFORE,
  STRDT: STRDT,
  STRENDS: STRENDS,
  STRLANG: STRLANG,
  STRLEN: STRLEN,
  STRSTARTS: STRSTARTS,
  STRUUID: STRUUID,
  SUBSTR: SUBSTR,
  SUM: SUM,
  SelectClause: SelectClause,
  SelectQuery: SelectQuery,
  ServiceGraphPattern: ServiceGraphPattern,
  SolutionModifier: SolutionModifier,
  SourceSelector: SourceSelector,
  StrReplaceExpression: StrReplaceExpression,
  String: String,
  SubSelect: SubSelect,
  SubstringExpression: SubstringExpression,
  TIMEZONE: TIMEZONE,
  TO: TO,
  TRUE: TRUE,
  TZ: TZ,
  TriplesBlock: TriplesBlock,
  TriplesNode: TriplesNode,
  TriplesNodePath: TriplesNodePath,
  TriplesSameSubject: TriplesSameSubject,
  TriplesSameSubjectPath: TriplesSameSubjectPath,
  TriplesTemplate: TriplesTemplate,
  UCASE: UCASE,
  UNDEF: UNDEF,
  UNION: UNION,
  URI: URI,
  USING: USING,
  UUID: UUID,
  UnaryExpression: UnaryExpression,
  Unit: Unit,
  Update: Update,
  Update1: Update1,
  UsingClause: UsingClause,
  VALUES: VALUES,
  ValueLogical: ValueLogical,
  ValuesClause: ValuesClause,
  Var: Var,
  VarOrIri: VarOrIri,
  VarOrTerm: VarOrTerm,
  Verb: Verb,
  VerbPath: VerbPath,
  VerbSimple: VerbSimple,
  WHERE: WHERE,
  WITH: WITH,
  WhereClause: WhereClause,
  YEAR: YEAR
});

/**
 * Case-insensitive keyword matcher for use as a Lezer @external specialize handler.
 *
 * SPARQL keywords are case-insensitive per the spec, with one exception: the
 * shorthand predicate 'a' (representing rdf:type) is case-sensitive and must
 * remain lowercase to avoid colliding with uppercase keyword matching.
 *
 * @param {string} value - The token text to look up
 * @param {Object} _stack - Lezer parse stack (required by the external specialize
 *   signature but not used here)
 * @returns {number} The term ID for the keyword, or -1 if not found
 */
const caseInsensitive = (value, _stack) => {
  return tokens[value.toUpperCase()] || -1;
};

// This file was generated by lezer-generator. You probably shouldn't edit it.
const spec_Keyword = {__proto__:null,a:634};
const parser = LRParser.deserialize({
  version: 14,
  states: "!A[O!TQPOOP!eOPOOO!jQPO'#EtO!oQPO'#EvOOQO'#IY'#IYO!tQPO'#EsO$pQPO'#HpQOQPOOPOOO'#IX'#IXP%QOPO)CDnOOQO,5;`,5;`O%XQPO,5;bOOQO-E<W-E<WO%^QPO'#EyO%oQPO'#ExO%zQPO'#HhO&VQPO'#HnO%oQPO'#HoO&hQPO,5;^O&pQPO'#HrO'OQPO'#HtO'aQPO'#HvO(TQPO'#HwO([QPO'#HyO(cQPO'#HzO(jQPO'#H{O(rQPO'#IVO(zQPO'#IUO)VQPO'#ITO)bQPO'#ITO&sQPO'#ITOOQO'#Hq'#HqO)jQPO,5>[POOO-E<V-E<VOOQO1G0|1G0|OOQO'#IZ'#IZO)rQPO'#IZO/[QPO,5;eOOQO,5;e,5;eO/mQPO,5;eOOQO'#Fe'#FeO%uQPO'#FeO1fQPO'#FcO2YQPO'#HdOOQO'#Is'#IsO%oQPO,5;dO2hQPO,5;dO3PQPO,5>SO4uQPO'#HiO%oQPO,5>SO4|QPO,5>SOOQO'#Fk'#FkOOQO'#Fj'#FjOOQO'#HU'#HUOOQO'#It'#ItO5RQPO,5>YO5lQPO,5>YO%oQPO,5>ZO2hQPO,5>ZO5yQPO'#FwOOQO1G0x1G0xO6UQPO,5>^O&sQPO,5>^O&sQPO'#HsOOQO'#Hu'#HuOOQO,5>`,5>`O6aQPO,5>`OOQO,5>b,5>bO6aQPO,5>bOOQO'#Hx'#HxO&sQPO'#HxO6oQPO,5>cO'rQPO,5>cO6tQPO,5>eO'rQPO,5>eO6yQPO,5>fO'rQPO,5>fOOQO,5>g,5>gO7OQPO,5>gO7TQPO,5>hO7YQPO'#ISOOQO,5>q,5>qO7TQPO,5>lO(uQPO,5>mOOQO,5>p,5>pO(uQPO'#IVO7dQPO'#IWOOQO'#Iv'#IvO7rQPO,5>oO%uQPO,5>oO7rQPO,5>oO7zQPO,5>oOYQPO1G3vO)rQPO'#FVOOQO'#FZ'#FZO8SQPO'#FXO8XQPO'#FXO8^QPO'#FXO8cQPO'#F_O8hQPO'#F`O8mQPO'#FaO9|QQO'#F{OOQO'#GP'#GPOOQO'#GS'#GSOOQO'#GW'#GWOOQO'#GO'#GOOOQO'#G['#G[O%uQPO'#FbO:jQPO'#HbOOQO'#FW'#FWO:oQPO'#FWO:tQPO'#FWO:yQPO'#FWO;OQPO'#FWO;WQPO'#FWO;]QPO'#FWO;eQPO'#FWO<tQQO'#HcOOQO'#FU'#FUO=bQPO'#FTOOQO'#FT'#FTOCoQQO'#FSOEZQQO'#FROOQO'#FQ'#FQOE}QQO'#FPOOQO'#FO'#FOOF{QPO'#E}OGaQPO'#E|OOQO'#E{'#E{OGrQPO,5>uOOQO-E<X-E<XOGwQPO1G1POOQO1G1P1G1POOQO,5<P,5<POHYQPO'#F{OJeQPO'#FdOOQO'#Gc'#GcOOQO'#Gb'#GbOOQO'#Ga'#GaO/xQPO'#GxOJmQPO'#G{OOQO'#Gw'#GwOJmQPO'#G`OKXQPO'#G`OK{QPO'#G_OLmQPO'#HYO%uQPO'#HZO%uQPO'#H[ON|QPO'#H]O! [QPO'#H^O! mQPO'#H_O!%zQPO'#H`O5yQPO'#HaOOQO'#HX'#HXO!&PQPO'#ImO!&sQPO'#G^O!&sQPO'#G^O!'bQPO,5;}OOQO'#Hf'#HfOOQO'#He'#HeO&sQPO'#HgOOQO,5>O,5>OOOQO-E<q-E<qO2hQPO1G1OO!'gQPO'#FgO! mQPO'#FnO!'lQPO'#FqO!'qQPO'#FtO!'qQPO'#FvO!'vQPO'#FsO!(UQPO'#FsOOQO'#Ff'#FfO!(dQPO'#FfO!(uQPO'#FfO!(|QPO'#FfOOQO1G1O1G1OO!)WQPO1G3nO3XQPO'#HQO!)]QPO'#HROOQO'#HP'#HPO!)]QPO'#HkO!)nQPO'#HkO!){QPO'#HjOOQO,5>T,5>TO!*TQPO,5>TO%oQPO1G3nO2hQPO1G3nO!*YQPO1G3nO5lQPO1G3tOOQO-E<r-E<rOOQO1G3t1G3tO2hQPO1G3tO2hQPO1G3uOOQO1G3u1G3uO!*aQPO'#FyO!*fQPO'#G]O!*kQPO'#G]OOQO'#Fx'#FxOOQO,5<c,5<cO7OQPO1G3xO!*sQPO1G3xOOQO,5>_,5>_OOQO1G3z1G3zOOQO1G3|1G3|OOQO,5>d,5>dO'rQPO1G3}O!+OQPO1G3}O'rQPO1G4PO!+TQPO1G4PO'rQPO1G4QO!+YQPO1G4QOOQO1G4R1G4RO7YQPO'#H}OOQO1G4S1G4SO!+_QPO'#HmON|QPO'#IPO!+jQPO'#IuO!+wQPO'#IOO!+wQPO'#IOO!,PQPO,5>nOOQO1G4W1G4WOOQO1G4X1G4XOOQO,5>r,5>rO&sQPO,5>rOOQO-E<t-E<tO%uQPO1G4ZOOQO1G4Z1G4ZO!,UQPO1G4ZO(uQPO'#IUO!,^QPO1G4ZO!,UQPO1G4ZO#{QPO'#HpOOQO7+)b7+)bO!,iQPO,5;qO!,nQPO,5;sO!,xQPO,5;sO!-PQPO,5;sO)rQPO,5;yO)rQPO,5;zO)rQPO,5;{OOQO,5<g,5<gO&sQPO,5<gOOQO,5;|,5;|O%uQPO,5=|O)rQPO,5;rO)rQPO,5;rO!-WQPO,5;rOOQO,5;r,5;rOOQO'#F^'#F^O)rQPO'#F^O)rQPO,5;rOOQO'#Fm'#FmO!-]QPO'#FmOOQO,5=},5=}OOQO,5;o,5;oO)rQPO'#IoO!.bQQO,5;nO)rQPO'#IpO!0SQQO'#IpO!1nQQO,5;mO!2bQPO,5;kO;]QPO,5;kO!2oQPO,5;kO)rQPO'#IqO!2tQPO,5;iO)rQPO'#IrO!3YQPO,5;hO!3kQPO1G4aO!4UQPO,5<OOOQO'#Ig'#IgO!4]QPO,5=dOOQO'#Gv'#GvOOQO'#Gn'#GnOOQO'#Gm'#GmO!4uQPO'#GmO!4|QPO'#GmO!5eQSO'#GlOOQO'#Gk'#GkO!7eQPO'#GkO!7yQSO'#GjO!9pQSO'#GiOOQO'#Gh'#GhOOQO'#Gg'#GgOOQO'#Gs'#GsO!;dQPO,5=gO/xQPO'#GfOOQO,5<z,5<zOOQO'#HV'#HVO!;iQPO,5<yO%uQPO'#InO!<YQPO,5=tOOQO,5=u,5=uOOQO,5=v,5=vO%uQPO,5=wO%uQPO,5=xON|QPO,5=xO!>iQPO'#FiOOQO'#Fp'#FpOOQO,5=y,5=yO)rQPO,5=zOOQO,5={,5={OOQO,5?X,5?XO!>qQPO,5?XOOQO-E<k-E<kO!?bQPO,5<xOOQO1G1i1G1iOOQO,5>R,5>ROOQO7+&j7+&jO!@PQPO,5<ROOQO'#Fo'#FoOOQO'#I^'#I^O!DaQPO,5<YO!DwQPO,5<]OOQO'#I`'#I`O!EUQPO,5<`O!EgQPO,5<bOOQO,5<_,5<_OOQO,5<Q,5<QO!ExQPO,5<QO!FZQPO,5<QO!FbQPO7+)YOOQO'#Ij'#IjO!FiQPO,5=lOOQO'#HO'#HOOOQO'#HT'#HTO3XQPO'#HSO!FpQPO,5=mOOQO,5>V,5>VOOQO'#Hl'#HlO!FuQPO,5>UOOQO1G3o1G3oO2hQPO7+)YOOQO7+)Y7+)YO!F|QPO7+)YOOQO7+)`7+)`O2hQPO7+)`OOQO7+)a7+)aO!H`QPO,5<eO!HgQPO,5<wOOQO'#Ib'#IbO!HrQPO,5<wO!HzQPO,5<wOOQO7+)d7+)dO7OQPO7+)dOOQO7+)i7+)iO'rQPO7+)iOOQO7+)k7+)kO'rQPO7+)kOOQO7+)l7+)lO'rQPO7+)lO!IPQPO,5>iO!IUQPO,5>XO!I`QPO,5>kOOQO,5?a,5?aO!IeQPO,5?aOOQO-E<s-E<sO!IoQPO,5>jOOQO1G4Y1G4YOOQO1G4^1G4^OOQO7+)u7+)uO%uQPO7+)uO!IwQPO7+)uO!IwQPO7+)uOOQO1G1]1G1]O!JPQPO1G1_O!JUQPO1G1_O)rQPO1G1_O!J]QPO1G1_O)rQPO1G1_O!JeQPO1G1eO!JjQPO1G1fO!JoQPO1G1gOOQO1G2R1G2ROOQO1G3h1G3hO!JtQPO1G1^O!JyQPO1G1^O!KOQPO,5;xO!KWQPO1G1^O!K]QPO,5<XO)rQPO,5<XOOQO,5?Z,5?ZOOQO-E<m-E<mOOQO,5?[,5?[O!LcQQO,5?[OOQO-E<n-E<nO;jQPO'#HcO!MVQPO'#FSO!MaQPO'#FROOQO1G1V1G1VO;]QPO1G1VOOQO,5?],5?]OOQO-E<o-E<oOOQO,5?^,5?^OOQO-E<p-E<pO!MkQPO7+){O!MpQPO1G1jOOQO-E<e-E<eOOQO1G3O1G3OOOQO'#Gq'#GqO!MxQPO'#GqOOQO'#Gp'#GpO!NWQPO'#GpOOQO,5=X,5=XO!N_QPO,5=XOOQO'#Gr'#GrOOQO,5=W,5=WOOQO,5=V,5=VO!4|QPO'#IeO!NdQSO,5=UO!4|QPO'#IfO#!ZQSO,5=TOOQO1G3R1G3ROOQO'#Gu'#GuO##}QPO'#GtO#$xQPO,5=QOOQO1G2e1G2eOOQO,5?Y,5?YOOQO-E<l-E<lOOQO1G3c1G3cOOQO1G3d1G3dO%uQPO1G3dOOQO,5<T,5<TO#%pQPO1G3fOOQO1G4s1G4sOOQO'#Fh'#FhO)rQPO'#FhOOQO'#I]'#I]O#%uQPO1G1mOOQO-E<[-E<[O!%uQPO'#FrOOQO'#Fr'#FrOOQO'#I_'#I_O#&`QPO1G1wOOQO-E<^-E<^OOQO1G1l1G1lO#&|QPO1G1lO2hQPO<<LtO#'_QPO<<LtOOQO-E<h-E<hOOQO1G3W1G3WO#'dQPO,5=nOOQO'#G}'#G}O#'uQPO'#G|OOQO1G3X1G3XOOQO1G3p1G3pOOQO<<Lt<<LtOOQO<<Lz<<LzOOQO'#Fz'#FzOOQO'#Ia'#IaO#(pQPO1G2POOQO1G2P1G2PO#(wQPO'#IcOOQO'#Ic'#IcO#)OQPO1G2cOOQO1G2c1G2cOOQO-E<`-E<`O#)ZQPO1G2cO#)OQPO1G2cOOQO<<MO<<MOOOQO<<MT<<MTOOQO<<MV<<MVOOQO<<MW<<MWOOQO1G4T1G4TOOQO1G3s1G3sO#)`QPO1G4VOOQO1G4{1G4{OOQO<<Ma<<MaO%uQPO<<MaO#)gQPO<<MaOOQO7+&y7+&yO#)oQPO7+&yO#)tQPO7+&yO#)yQPO7+&yO)rQPO7+'PO)rQPO7+'QO)rQPO7+'ROOQO7+&x7+&xO)rQPO7+&xO)rQPO'#I[O#*RQPO1G1dOOQO1G1d1G1dO)rQPO7+&xO#*ZQPO1G1sOOQO1G1s1G1sO#*ZQPO1G1sO#*cQPO,5;nO#*mQPO'#IpO#*wQPO,5;mOOQO7+&q7+&qOOQO<<Mg<<MgOOQO7+'U7+'UOOQO,5=],5=]OOQO,5=[,5=[O#+RQSO,5=[OOQO1G2s1G2sOOQO,5?P,5?POOQO-E<c-E<cOOQO,5?Q,5?QOOQO-E<d-E<dO/xQPO'#IhO#+ZQPO,5=`O#,UQPO'#IiO#-OQPO1G2lOOQO7+)O7+)OO#-vQPO7+)QO#-{QPO,5<SOOQO-E<Z-E<ZOOQO,5<^,5<^OOQO-E<]-E<]OOQO7+'W7+'WOOQOANB`ANB`O2hQPOANB`O#.TQPO'#IkO#.hQPO1G3YO3XQPO'#IlO#.yQPO,5=hOOQO-E<_-E<_OOQO7+'k7+'kO#/tQPO,5>}OOQO,5>},5>}OOQO-E<a-E<aOOQO7+'}7+'}O#/{QPO7+'}O#/{QPO7+'}OOQO7+)q7+)qO#0WQPO7+)qOOQOANB{ANB{O%uQPOANB{OOQO<<Je<<JeO#0]QPO<<JeO#0bQPO<<JeO#0gQPO<<JkO#0oQPO<<JlO#0tQPO<<JmO#0|QPO<<JdOOQO,5>v,5>vOOQO-E<Y-E<YOOQO7+'O7+'OO#1RQPO<<JdOOQO7+'_7+'_O#1WQPO7+'_O#1`QPO,5?[O!4dQPO'#IdO#1jQSO1G2vOOQO1G2v1G2vOOQO,5?S,5?SOOQO-E<f-E<fO3XQPO,5?TOOQO-E<g-E<gO#1rQPO<<LlOOQO1G1n1G1nO#1wQPO1G1nOOQOG27zG27zO3XQPO,5?VOOQO-E<i-E<iOOQO,5?W,5?WOOQO-E<j-E<jOOQO1G4i1G4iO#1|QPO<<KiOOQO<<Ki<<KiOOQO<<M]<<M]OOQOG28gG28gO1WQPOAN@PO#2XQPOAN@POOQOAN@VAN@VO)rQPOAN@VO)rQPOAN@WOOQOAN@XAN@XO)rQPOAN@XOOQOAN@OAN@OO)rQPOAN@OOOQO<<Jy<<JyOOQO,5?O,5?OOOQO-E<b-E<bOOQO7+(b7+(bOOQO1G4o1G4oOOQOANBWANBWO#2^QPO7+'YOOQO1G4q1G4qOOQOANATANATO#2cQPOG25kO1WQPOG25kO#2hQPOG25qO#2mQPOG25rO#2uQPOG25sO#2zQPOG25jOOQO<<Jt<<JtOOQOLD+VLD+VO#3PQPOLD+VOOQOLD+]LD+]OOQOLD+^LD+^O)rQPOLD+^OOQOLD+_LD+_OOQOLD+ULD+UOOQO!$'Nq!$'NqO#3UQPO!$'NxOOQO!)9Dd!)9DdO=bQPO'#FTO!2bQPO'#IoO!2bQPO'#Ip",
  stateData: "#3q~O'mOS#dPQ~OPQOQRO#S#gP#V#gP#W#gP#X#gP#Y#gP#Z#gP#[#gP#]#gP#_#gP#`#gP'k#gP~OR#gPS#gPT#gP]#gP~PYO#dWO~O#iYO~O#kZO~OPQOQROR#gXS#gXT#gX]#gX#S#gX#V#gX#W#gX#X#gX#Y#gX#Z#gX#[#gX#]#gX#_#gX#`#gX'k#gX~O#ScO#VdO#WeO#XiO#YfO#ZgO#[hO#]jO#_kO#`nO'k&dX~OR]OS_OT`O]aO~P#{O#dWO~Q'l]#irO~OVwOWwO#nsO'ntO'qvO~OUyO#O{O'xzO~OU!SO#O{O'x!QO~O#i!UO#k!TO#n!VO$`!TO'q!YO~O!s!]O'k$kP~O!z!`O#i!UO#k!TO$`!TO~O!x!aO!z!dO#P!bO#a!bO#b!bO~O!x!aO!z!fO#P!bO#a!bO#b!bO~O!x!hO#a!gO#i!UO#k!TO$`!TO~O!z!jO~P'rO!z!lO~P'rO!z!nO~P'rO!x!aO!z!pO~O#^!qO'x!rO~OU!uO#^!tO'x!rO~OU!{O#]!wO#c!xO~OU!{O#c!xO~O#|#OO'k&da~OX#bOY#bOZ#cO[#bO^#dO_#bO`#bOa#eOb#fOc#bOd#bOe#bOf#bOg#gOh#bOi#bOj#bOk#bOl#cOm#cOn#cOo#cOp#cOq#bOr#bOs#bOt#bOu#bOv#bOw#bOx#bOy#fOz#fO{#fO|#bO}#bO!O#bO!P#bO!Q#bO!R#gO!S#hO!T#cO!U#cO!V#cO!W#bO!X#bO!Y#bO!Z#bO![#bO!]#RO!^#SO!_#SO!`#SO!a#SO!b#SO!c#TO!e#UO!f#VO!g#WO!h#_O!i#`O#Q#^O#R#^O#i!UO#k!TO#n#jO$`!TO$i#YO$t#YO$u#YO$w#ZO$x#ZO$y#ZO${#[O$|#[O$}#[O'n#PO'o#kO's#QO't#QO'u#QO'v#QO'}#kO(P#kO~O#nsO'ntOU#ma#O#ma'x#ma~O#nsO'ntO'q#xO~O#Q#^O#R#^O#i!UO#k!TO#n$OO$P#}O$`!TO$i#YO$t#YO$u#YO$w#ZO$x#ZO$y#ZO${#[O$|#[O$}#[O%W#|O%X#|O%n$QO'n$PO's#QO't#QO'u#QO'v#QO~OR]O!s$^O!v$WO!w$XO!x$YO!y$ZO!{$[O!|$]O'xzO'w%QP~P/xO#P$fO#i!UO#k!TO$`!TO~O!k$jO!m$kO!n$lO!q$mO!r$nO!s$YP'k$YP~OU$vO#O{O~O#Q#^O#R#^O#i!UO#k!TO#n$OO$P#}O$`!TO$i#YO$t#YO$u#YO$w#ZO$x#ZO$y#ZO${#[O$|#[O$}#[O%W#|O%X#|O%n$xO'n$wO's#QO't#QO'u#QO'v#QO~O'w$}O~P3XO'x%RO~OUyO#O{O#i!UO#k!TO#n!VO$`!TO'xzO~P2hOUyO#O{O'xzO~P2hO#n%YO$P%ZO'n%[O~O#T%_O#|&fa'k&fa~O!x!aO#P!bO#a!bO#b!bO~O#U%eO~O#U%gO~O#U%iO~O!x!aO~O'x%lO~O!x%oO'w&rP~P3XO#P%wO#i!UO#k!TO$`!TO~OU%yO#c!xO~O#]!wO#_%|O~O'n&SO~O'n&TO~O'n&UO~O'n&VO~O'n&WO~O'n&XO~O$p&YO$q&ZO!}$oX$w$oX$x$oX$y$oX${$oX$|$oX$}$oX'q$oX'}$oX(O$oX(P$oX(V$oX(W$oX'p$oX#|$oX$O$oX~O!i$oX!j$oX'r$oX(Q$oX(R$oX(S$oX(T$oX(U$oX~P8rO!h&]O~O'n&^O~O'n&_O~O'n&`O~O$P&aO'n&^O~O$P&aO~O$P&bO'n&cO~O'n&dO~O$P&eO'n&fO!}&VX$w&VX$x&VX$y&VX${&VX$|&VX$}&VX'q&VX'}&VX(O&VX(P&VX(V&VX(W&VX'p&VX#|&VX$O&VX~O!i&VX!j&VX'r&VX(Q&VX(R&VX(S&VX(T&VX(U&VX~P;jOX#bOY#bOZ#cO[#bO^#dO_#bO`#bOa#eOb#fOc#bOd#bOe#bOf#bOg#gOh#bOi#bOj#bOk#bOl#cOm#cOn#cOo#cOp#cOq#bOr#bOs#bOt#bOu#bOv#bOw#bOx#bOy#fOz#fO{#fO|#bO}#bO!O#bO!P#bO!Q#bO!R#gO!S#hO!T#cO!U#cO!V#cO!W#bO!X#bO!Y#bO!Z#bO![#bO!]#RO!^#SO!_#SO!`#SO!a#SO!b#SO!c#TO!e#UO!f#VO!g#WO!h#_O!i#`O#Q#^O#R#^O#i!UO#k!TO#n#jO$`!TO$i#YO$t#YO$u#YO$w#ZO$x#ZO$y#ZO${#[O$|#[O$}#[O'n#PO's#QO't#QO'u#QO'v#QO~O!}#vX$w#vX$x#vX$y#vX${#vX$|#vX$}#vX'}#vX(P#vX(V#vX(W#vX'p#vX#|#vX$O#vX~O'q&iO(O&iO!i#vX!j#vX'r#vX(Q#vX(R#vX(S#vX(T#vX(U#vX~PBqO$w#ZO$x#ZO$y#ZO${#[O$|#[O$}#[O!}#uX(V#uX(W#uX'p#uX#|#uX$O#uX~O'}&kO(P&kO!i#uX!j#uX'r#uX(Q#uX(R#uX(S#uX(T#uX(U#uX~PDcO!i&pO!j&oO'r&nO(Q&nO(R&nO(S&nO(T&nO(U&nO!}#sX(V#sX(W#sX'p#sX#|#sX$O#sX~O(V&qO!}#qX(W#qX'p#qX#|#qX$O#qX~O(W&sO!}#pX'p#pX#|#pX$O#pX~O!}&uO~O#nsO'ntOU#mi#O#mi'x#mi~O#i$oX#k$oX#n$oX$`$oX'n$oX'o$oX'y$oX'z$oX#Q$oX#R$oX$P$oX$i$oX$t$oX$u$oX%W$oX%X$oX%n$oX's$oX't$oX'u$oX'v$oX%m$oX!s$oX!v$oX!w$oX!x$oX!y$oX!{$oX!|$oX%z$oX'w$oX'x$oX!t$oX~P8rOUyO'xzO~O#i!UO#k!TO#n'VO$`!TO'n&}O'o&|O'y&zO'z'QO~O!s%yP!v%yP!w%yP!x%yP!y%yP!{%yP!|%yP%z%yP'w%yP'x%yP~PJmO%z'[O!s%RX!v%RX!w%RX!x%RX!y%RX!{%RX!|%RX'w%RX'x%RX~O!u']O!s%|X!v%|X!w%|X!x%|X!y%|X!{%|X!|%|X#Q%|X#R%|X#i%|X#k%|X#n%|X$P%|X$`%|X$i%|X$t%|X$u%|X$w%|X$x%|X$y%|X${%|X$|%|X$}%|X%W%|X%X%|X%n%|X%z%|X'n%|X's%|X't%|X'u%|X'v%|X'w%|X'x%|X~O#i!UO#k!TO#n!VO$`!TO~O!z'cO#i!UO#k!TO#n!VO$`!TO~OX#bOY#bOZ#cO[#bO^#dO_#bO`#bOa#eOb#fOc#bOd#bOe#bOf#bOg#gOh#bOi#bOj#bOk#bOl#cOm#cOn#cOo#cOp#cOq#bOr#bOs#bOt#bOu#bOv#bOw#bOx#bOy#fOz#fO{#fO|#bO}#bO!O#bO!P#bO!Q#bO!R#gO!S#hO!T#cO!U#cO!V#cO!W#bO!X#bO!Y#bO!Z#bO![#bO!]#RO!^#SO!_#SO!`#SO!a#SO!b#SO!c#TO!e#UO!f#VO!g#WO!h#_O!i#`O#i!UO#k!TO$`!TO'n#PO~O'n'gO~O%z'jO!s'aX!v'aX!w'aX!x'aX!y'aX!{'aX!|'aX'w'aX'x'aX~P/xO!s$^O!v$WO!w$XO!x$YO!y$ZO!{$[O!|$]O'xzO'w%QX~O'w'mO~O!l'pO~O!l'tO~O$i'uO~O!r$nO!s$gX'k$gX'w$gX~O!q$mO!s$gX'k$gX'w$gX~O!q$mO!r$nO!s$YX'k$YX'w$YX~O!n$lO~P!(dO!m$kO!n$lO~P!(dO'x'|O~O#i!UO#k!TO#n!VO$`!TO'y&zO~O%z&`P'w&`P!x&`P~P!)]O%z(VO'w&^X~O'w(WO~O'w(XO~P3XO'x(_O~O'x(`O~O#n(aO'p(cO~O#T(eO#|&fi'k&fi~O#U(gO~O#U(iO~O#U(kO~O%z(mO!x&aX'w&aX~O%z(pO!x'iX'w'iX~P3XO!x%oO'w&rX~O'w(sO~OU(vO#c!xO~OU(vO#]!wO#c!xO~O'p(yO~OV({O'q(zO~P)rOV(|O~P)rOV)OO~P)rO#n)UO~OV)ZO~P)rO!}#va$w#va$x#va$y#va${#va$|#va$}#va'}#va(P#va(V#va(W#va'p#va#|#va$O#va~O'q&iO(O&iO!i#va!j#va'r#va(Q#va(R#va(S#va(T#va(U#va~P!-dO!}'dX$w'dX$x'dX$y'dX${'dX$|'dX$}'dX'}'dX(P'dX(V'dX(W'dX'p'dX#|'dX$O'dX~O'q&iO(O&iO!i'dX!j'dX'r'dX(Q'dX(R'dX(S'dX(T'dX(U'dX~P!/UO$w#ZO$x#ZO$y#ZO${#[O$|#[O$}#[O!}#ua(V#ua(W#ua'p#ua#|#ua$O#ua~O'}&kO(P&kO!i#ua!j#ua'r#ua(Q#ua(R#ua(S#ua(T#ua(U#ua~P!0vO'o-}O'}-}O(P-}O~P=bO!j)eO~O(V&qO!}#qa(W#qa'p#qa#|#qa$O#qa~O(W&sO!}#pa'p#pa#|#pa$O#pa~O#n)jO~O!k$jO!m$kO!n$lO!q$mO!r$nO!s$YP~O'w$YP~P!3pO'p)mO~P/xO#i!UO#k!TO$`!TO'y&zO'z)oO~O'n)qO~P!4dO#i!UO#k!TO$`!TO'n&}O'o&|O'y&zO'z'QO~O'q)tO'|)tO'})tO#Q%`X#R%`X#i%`X#k%`X#n%`X$P%`X$`%`X$i%`X$t%`X$u%`X$w%`X$x%`X$y%`X${%`X$|%`X$}%`X%W%`X%X%`X%n%`X'n%`X's%`X't%`X'u%`X'v%`X'{%`X(O%`X'p%`X~O#i!UO#k!TO$`!TO'n&}O'o&|O'y&zO~O(O)wO#Q%^X#R%^X#i%^X#k%^X#n%^X$P%^X$`%^X$i%^X$t%^X$u%^X$w%^X$x%^X$y%^X${%^X$|%^X$}%^X%W%^X%X%^X%n%^X'n%^X's%^X't%^X'u%^X'v%^X'{%^X'p%^X~O'{)yO#Q%]X#R%]X#i%]X#k%]X#n%]X$P%]X$`%]X$i%]X$t%]X$u%]X$w%]X$x%]X$y%]X${%]X$|%]X$}%]X%W%]X%X%]X%n%]X'n%]X's%]X't%]X'u%]X'v%]X'p%]X~O%m){O~O!s%Ra!v%Ra!w%Ra!x%Ra!y%Ra!{%Ra!|%Ra'w%Ra'x%Ra~P/xO!u']O!s%|a!v%|a!w%|a!x%|a!y%|a!{%|a!|%|a#Q%|a#R%|a#i%|a#k%|a#n%|a$P%|a$`%|a$i%|a$t%|a$u%|a$w%|a$x%|a$y%|a${%|a$|%|a$}%|a%W%|a%X%|a%n%|a%z%|a'n%|a's%|a't%|a'u%|a'v%|a'w%|a'x%|a~O$P&eO'n&fO~O!s'aa!v'aa!w'aa!x'aa!y'aa!{'aa!|'aa'w'aa'x'aa~P/xO!s$^O!v$WO!w$XO!x$YO!y$ZO!{$[O!|$]O'xzO'w%Qa~OX#bOY#bOZ#cO[#bO^#dO_#bO`#bOa#eOb#fOc#bOd#bOe#bOf#bOg#gOh#bOi#bOj#bOk#bOl#cOm#cOn#cOo#cOp#cOq#bOr#bOs#bOt#bOu#bOv#bOw#bOx#bOy#fOz#fO{#fO|#bO}#bO!O#bO!P#bO!Q#bO!R#gO!S#hO!T#cO!U#cO!V#cO!W#bO!X#bO!Y#bO!Z#bO![#bO!]#RO!^#SO!_#SO!`#SO!a#SO!b#SO!c#TO!e#UO!f#VO!g#WO!h#_O!i#`O#i!UO#k!TO#n*YO$`!TO'n*ZO~O!n$ba!q$ba!r$ba!s$ba'k$ba'w$ba~P! mO!o*_O!p*_O#n*`O~P! mO$i'uO!r$ha!s$ha'k$ha'w$ha~O$i'uO!q$ja!s$ja'k$ja'w$ja~O!q$mO!r$nO!s$Ya'k$Ya'w$Ya~O!n$lO~P!ExO'w*fO~P3XO'p*iO~P3XO%m*mO~O'w&^a~P3XO'w*fO~O!t*qO#Q#^O#R#^O#i!UO#k!TO$`!TO$i#YO$t#YO$u#YO$w#ZO$x#ZO$y#ZO${#[O$|#[O$}#[O's#QO't#QO'u#QO'v#QO~O'w*tO~P!GRO$P*vO'n*uO'w*xO~O#n(aO'p*zO~O'x*{O~O'w+QO~O!x&aa'w&aa~P3XO'x+SO~O!x'ia'w'ia~P3XO!x%oO'w&ra~OU+VO#c!xO~O'p+XO~O'q+YO~P)rO#|+ZO'p+XO~O$O+]O~O$O+^O~O$O+_O~O'p+`O~O$O+aO~O$O+bO'p+dO~O$O+eO~O$O+bO'p+gO~O!}'da$w'da$x'da$y'da${'da$|'da$}'da'}'da(P'da(V'da(W'da'p'da#|'da$O'da~O'q&iO(O&iO!i'da!j'da'r'da(Q'da(R'da(S'da(T'da(U'da~P!KeO'q.OO(O.OO~PBqO'}.PO(P.PO~PDcO'p+mO~O!s!]O'w$kP~O#i!UO#k!TO$`!TO'y&zO~O'p+pO~P!4dO'p+rO~O(O)wO#Q%^a#R%^a#i%^a#k%^a#n%^a$P%^a$`%^a$i%^a$t%^a$u%^a$w%^a$x%^a$y%^a${%^a$|%^a$}%^a%W%^a%X%^a%n%^a'n%^a's%^a't%^a'u%^a'v%^a'{%^a'p%^a~O'{)yO#Q%]a#R%]a#i%]a#k%]a#n%]a$P%]a$`%]a$i%]a$t%]a$u%]a$w%]a$x%]a$y%]a${%]a$|%]a$}%]a%W%]a%X%]a%n%]a'n%]a's%]a't%]a'u%]a'v%]a'p%]a~O$O+wO#|%hX%m%hX!s%hX!v%hX!w%hX!x%hX!y%hX!{%hX!|%hX%z%hX'w%hX'x%hX~O#|+yO%m%Ya!s%Ya!v%Ya!w%Ya!x%Ya!y%Ya!{%Ya!|%Ya%z%Ya'w%Ya'x%Ya~O!}+|O~O!m$Zi!n$Zi!q$Zi!r$Zi!s$Zi'k$Zi'w$Zi~P!@PO!o*_O!p*_O#n*`O!q$ei!r$ei!s$ei'k$ei'w$ei~P! mO!q$mO!r$nO!s$Yi'k$Yi'w$Yi~O'w,TO~O#|,UO%m%va%z%va'w%va!x%va~O$O,WO#|%pX%m%pX%z%pX'w%pX!x%pX!s%pX!v%pX!w%pX!y%pX!{%pX!|%pX'x%pX~O'w,ZO~P!GRO'p,]O~P!GRO$P*vO'n*uO'w,_O~O'x,`O~O'w,bO~P3XOU,eO#c!xO~O'p,fO~O!d,gO~O#|,hO'p,fO~O$O+bO'p,oO~O$O+bO'p,qO~O'q.OO(O.OO~P!-dO'q.OO(O.OO~P!/UO'}.PO(P.PO~P!0vO'p,vO'{,tO~O$O+wO#|%ha%m%ha!s%ha!v%ha!w%ha!x%ha!y%ha!{%ha!|%ha%z%ha'w%ha'x%ha~O#|']X%m']X!s']X!v']X!w']X!x']X!y']X!{']X!|']X%z']X'w']X'x']X~PJmO#|+yO%m%Yi!s%Yi!v%Yi!w%Yi!x%Yi!y%Yi!{%Yi!|%Yi%z%Yi'w%Yi'x%Yi~O#n,{O~O!},}O'p,|O~O#|'_X%m'_X%z'_X'w'_X!x'_X~P!)]O#|,UO%m%vi%z%vi'w%vi!x%vi~O$O,WO#|%pa%m%pa%z%pa'w%pa!x%pa!s%pa!v%pa!w%pa!y%pa!{%pa!|%pa'x%pa~O'p-TO~P!GRO$P*vO'n*uO'w-VO~O'w-WO~O'r-YO~O!d-ZO~O$O-]O'p-[O~O$O-^O~O$O-`O'p-_O~O'p-aO~O$O-bO~O$O+bO'p-cO~O'q.OO(O.OO~P!KeO'p-fO'{,tO~O'p-hO~O#n-iO~O$P*vO'n*uO'w-kO~O'r-mO~O'p-rO~O'p-sO~O'p-uO~O$O-wO'p-vO~O'p-xO~O'p-yO~O'p-zO~O'p-|O~O$`#k%W%c$}$|${'m#n'|$y$x$w$u$t$i(P$x'}%z'}~",
  goto: "!-g'kPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPP'l'o'uP'uP'z'}P(T)}*o+d,Y,|-s.h/c0`1]2b3hP4jPP6Z3h3h3h3h6d7r7u8c9Z9k3]9o=RP?o?v@Z@_@jARAVAsPB^BwB}CTCXC_PPC_DzPPFcPPPFcPPPC_CT7rHSH`HfIWIrPPJ^JgJoJxKPKZKfKrK|PLfLiLrJgLuLxMOMWMdPPMdMnMwNPNZNoNo! R! [! d! zP! }!!T!!T!!T!!T!!T!!T!!T!!T3h!!Z!#U!#c!#f!#c'z!#l!#o!#u!$T!$W'z'z!$n!$t!$x!$|!$x!%`!$x!$x!%l!$x!$x!$x!$x!&b!&h!&n!$x!$x!&u!$x!'Q!'X!'f!'r!'x!(P!(Z!(i!(o!(u!({!)V!)a!)g!)u!){!*R!*X!*_!*e!*k!*q!*w!*}!+X!+_!+q!+{!,R!,X!,p!,v!-RRVOQUOR&P#OVSOT#ORbUQ^UR#{zQ#utQ&R#PS(z&S&TQ(}&UQ)P&VQ)Q&WQ)R&XQ)U&^Q)V&_Q)W&cQ)X&dQ)Y&fQ*W'gS+Y({(|Q+[)OQ+h)ZQ+}*ZQ,i+]Q,j+^Q,k+_Q,l+aQ,m+bQ,p+eQ-n-]Q-o-^Q-p-`Q-q-bR-{-w!_#tt#P&S&T&U&V&W&X&^&_&c&d&f'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w!^#st#P&S&T&U&V&W&X&^&_&c&d&f'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-wR)h&s!`#rt#P&S&T&U&V&W&X&^&_&c&d&f&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-wR)f&q!c#qt#P&S&T&U&V&W&X&^&_&c&d&f&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w!b#pt#P&S&T&U&V&W&X&^&_&c&d&f&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-wR)d&n!e#ot#P&S&T&U&V&W&X&^&_&c&d&f&n&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w!b#nt#P&S&T&U&V&W&X&^&_&c&d&f&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-wS)^&k.PR)c&n!d#mt#P&S&T&U&V&W&X&^&_&c&d&f&k&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-wS)[&i.OT)b&n.P!l#lt#P&S&T&U&V&W&X&^&_&c&d&f&i&k&n&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w.O.PT&h#k-}!p#jt#P#k&S&T&U&V&W&X&^&_&c&d&f&i&k&n&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w-}.O.PY'e$[$k's't*bR,P*_!p#jt#P#k&S&T&U&V&W&X&^&_&c&d&f&i&k&n&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w-}.O.PY'e$[$k's't*bT*Y'p*]#P#at#P#k$[$k&S&T&U&V&W&X&^&_&c&d&f&i&k&n&q&s'g'p's't({(|)O)Z*Z*]*b+]+^+_+a+b+e-]-^-`-b-w-}.O.P!h#Xt#P#k&S&T&U&V&W&X&^&_&c&d&f&i&k&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w!b#zz!Q!r$P$`$w%R%l%p&n&x'X'['j'|(O(R(V(_(m(p*s*u+S+w,W,[,y-P-}.O.PQ-l-YR-t-mQ&a#gQ)d&oR+l)edx^a}!R!X!Y!Z#{%P%SQ#yyW$Vz$a$b'lQ%z!{Q&[#_Q'_$WQ'`$XQ(u%yQ)T&]Q*Q']Q*S'aQ*T'bQ+U(vQ+{*UQ,d+VR-X,eR$czQ!O^Q![aQ$i}Q%Q!RS%V!X!YQ%W!ZQ&v#{Q(X%PR(]%SQ$u!OS%U!X!YQ%X![Q'o$iQ(Y%QS([%S%VQ(^%WQ)k&vQ*o(XQ*p(]Q,S*fR-O,Tm$t!O!X!Y![$i%Q%S%V%W&v(X(]*f,TT*['p*]d!V`!X$Y$Z$x$z${%o'c,UQ!_ch!gfgh!j!l!n%e%g%i(g(i(kQ!}n!h#it#P#k&S&T&U&V&W&X&^&_&c&d&f&i&k&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w!Q#}z!Q!r$P$`$w%R%l%p&x'X'['j'|(O(R(V(m(p+S+w,W,y-PS$d{$fQ%`!`Q%a!aQ%d!hQ%v!x`&{$Q$S$T&}'Q)w)y+y^'d$[$k'p's't*]*bQ(t%wQ)S&ZW)a&n-}.O.PU)n&|)q,tW*q(_*s*u,[R+o)o&V!U`cfghntz{!Q!X!`!a!h!j!l!n!r!x#P#k$P$Q$S$T$Y$Z$[$`$f$k$w$x$z${%R%e%g%i%l%o%p%w&S&T&U&V&W&X&Z&^&_&c&d&f&i&k&n&q&s&x&|&}'Q'X'['c'g'j'p's't'|(O(R(V(_(g(i(k(m(p({(|)O)Z)o)q)w)y*Z*]*b*s*u+S+]+^+_+a+b+e+w+y,U,W,[,t,y-P-]-^-`-b-w-}.O.PS&g#i)aR*V'dl$s!O!X!Y![$i%Q%S%V%W&v(X(]*f,TR'{$tT'r$k'sQ'f$[S'q$k'sT*`'t*bl$r!O!X!Y![$i%Q%S%V%W&v(X(]*f,TS'z$s$tR*e'{T*a't*bl$q!O!X!Y![$i%Q%S%V%W&v(X(]*f,TU'y$r$s$tS*d'z'{R,R*ex$o!O!X!Y![$i$r$s$t%Q%S%V%W&v'z'{(X(]*e*f,TR'x$px$p!O!X!Y![$i$r$s$t%Q%S%V%W&v'z'{(X(]*e*f,TR'x$oQ!^bR+n)kQ%^!]R'h$^T%]!]$^X*r(_*s*u,[!p#jt#P#k&S&T&U&V&W&X&^&_&c&d&f&i&k&n&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w-}.O.P!Q#}z!Q!r$P$`$w%R%l%p&x'X'['j'|(O(R(V(m(p+S+w,W,y-PX*q(_*s*u,[#{#]tz!Q!r#P#k$P$`$w%R%l%p&S&T&U&V&W&X&^&_&c&d&f&i&k&n&q&s&x'X'['g'j'|(O(R(V(_(m(p({(|)O)Z*Z*s*u+S+]+^+_+a+b+e+w,W,[,y-P-]-^-`-b-w-}.O.P#z#]tz!Q!r#P#k$P$`$w%R%l%p&S&T&U&V&W&X&^&_&c&d&f&i&k&n&q&s&x'X'['g'j'|(O(R(V(_(m(p({(|)O)Z*Z*s*u+S+]+^+_+a+b+e+w,W,[,y-P-]-^-`-b-w-}.O.PS&l#n&mT+j)c+kQ$bzQ'i$`Q*P'[R*X'jX$Uz$`'['jW$Sz$`'['jd$z!Q!r%R%l%p'|(V(m(p+SW&y$P&x'X+w](P$w(O(R,W,y-P!R$Oz!Q!r$P$`$w%R%l%p&x'X'['j'|(O(R(V(m(p+S+w,W,y-P!R#}z!Q!r$P$`$w%R%l%p&x'X'['j'|(O(R(V(m(p+S+w,W,y-PQ'W$QQ'Y$SR'Z$TU'X$Q$S$TR,y+yW'U$Q$S$T+yR)s&}Z'T$Q$S$T&}+yY'S$Q$S$T&}+yR+u)y['R$Q$S$T&})y+yR+s)w^'P$Q$S$T&})w)y+yR)v'Qa'O$Q$S$T&}'Q)w)y+y`&{$Q$S$T&}'Q)w)y+yW(Q$x$z${,UU)n&|)q,tR+o)oR)r&|Q)p&|Q+q)qR-d,tR)u'OR*O'XQ)}'XR,w+wS&w$P&xT)|'X+wW$Tz$`'['jX&y$P&x'X+wa$Rz$P$`&x'X'['j+wQ*j(RQ-g,yR-j-PU*l(R,y-PR-R,WS'}$w(OX*k(R,W,y-Pd${!Q!r%R%l%p'|(V(m(p+S](P$w(O(R,W,y-Pq$y!Q!r$w%R%l%p'|(O(R(V(m(p+S,W,y-PQ(S$xQ(T$zR(U${U(R$x$z${R-P,US!W`!XQ'a$YQ'b$ZW(Q$x$z${,UQ(n%oR*U'cR'Y$TX$`z$a$b'lX$_z$a$b'l!q#jt#P#k&S&T&U&V&W&X&^&_&c&d&f&i&k&n&q&s'g({(|)O)Z*Z+]+^+_+a+b+e-]-^-`-b-w-}.O.Pg|^_a}!P!R!X!Y!Z%P%SR$g{Q$e{R'n$fR!R_Q%O!QR*n(VS$|!Q(Va%n!r%R%l%p'|(m(p+SR(T${S%r!r%lQ(Z%RQ(o%pQ*g'|Q+R(mQ+T(pR,c+SQVOR&Q#OTpU&PToU&PW!bde!d!fQ!oiQ%k!pQ(d%_R*|(eQ!cdQ!eeQ%b!dR%c!fQ!ifQ!kgQ!mhQ%f!jQ%h!lQ%j!nQ(f%eQ(h%gQ(j%iQ*}(gQ+O(iR+P(kQ%m!qR%t!tQ%s!rR(l%lZ%p!r%l%q%r(rS!sj!wS!vk%|R%u!uSlU&PR%}!}SmU&PQ!|lQ&O!}R(x%}e!ylm!z!|%{%}&O(w(x+WQXPRqXSTO#OR[TQu]S#vu#wR#wwQ+c)WQ+f)YU,n+c+f,rR,r+hQ*]'pR,O*]Q's$kR*^'sQ*b'tR,Q*bQ'v$mQ'w$nT*c'v'wQ*s(_S,Y*s,[R,[*uQ(b%[R*y(bQ*w(`U,^*w,a-UQ,a*{R-U,`Q,u+qR-e,uQ)x'RR+t)xQ)z'SR+v)zQ&x$PR)l&xQ+x)}R,x+xQ+z*OR,z+zQ(O$wR*h(OQ,V*jR-Q,VQ,X*lR-S,XQ$azS'k$a'lR'l$bQ'^$VR*R'^Q&j#mW)]&j)_+i,sQ)_&lQ+i)bR,s+jQ&m#nS)`&m+kR+k)cQ&r#rR)g&rQ&t#sR)i&tQ}^Q!P_Q!ZaY$h}!P!Z%P%SQ%P!RT%S!X!YQ!X`R%T!XS%q!r%lS(q%q(rR(r%rS!zlmW%x!z%{(w+WQ%{!|S(w%}&OR+W(x",
  nodeNames: "⚠ BASE PREFIX SELECT CONSTRUCT DESCRIBE WHERE DISTINCT REDUCED STR LANG LANGMATCHES DATATYPE ASK BOUND IRI URI BNODE RAND ABS CEIL FLOOR ROUND CONCAT STRLEN UCASE LCASE ENCODE_FOR_URI CONTAINS STRSTARTS STRENDS STRBEFORE STRAFTER YEAR MONTH DAY HOURS MINUTES SECONDS TIMEZONE TZ NOW UUID STRUUID MD5 SHA1 SHA256 SHA384 SHA512 COALESCE IF STRLANG STRDT SAMETERM ISIRI ISURI ISBLANK ISLITERAL ISNUMERIC COUNT SUM MIN MAX AVG SAMPLE GROUP_CONCAT SEPARATOR SUBSTR REPLACE REGEX EXISTS NOT IN GROUP BY HAVING ORDER ASC DESC LIMIT OFFSET VALUES UNDEF UNION OPTIONAL MINUS GRAPH SERVICE SILENT FILTER BIND AS FROM NAMED TRUE FALSE LOAD INTO TO CLEAR DROP CREATE ADD MOVE COPY INSERT DATA DELETE WITH DEFAULT ALL USING Comment Unit Query Prologue BaseDecl IriRef PrefixDecl Pname_ns SelectQuery SelectClause Var Expression ConditionalOrExpression ConditionalAndExpression ValueLogical RelationalExpression NumericExpression AdditiveExpression MultiplicativeExpression UnaryExpression PrimaryExpression BrackettedExpression BuiltInCall Aggregate ; String , Nil ExpressionList SubstringExpression StrReplaceExpression RegexExpression ExistsFunc GroupGraphPattern SubSelect WhereClause SolutionModifier GroupClause GroupCondition FunctionCall Iri PrefixedName Pname_ln ArgList HavingClause HavingCondition Constraint OrderClause OrderCondition LimitOffsetClauses LimitClause Integer OffsetClause ValuesClause DataBlock InlineDataOneVar DataBlockValue RDFLiteral Langtag ^^ NumericLiteral NumericLiteralUnsigned Decimal Double NumericLiteralPositive IntegerPositive DecimalPositive DoublePositive NumericLiteralNegative IntegerNegative DecimalNegative DoubleNegative BooleanLiteral InlineDataFull GroupGraphPatternSub TriplesBlock TriplesSameSubjectPath VarOrTerm GraphTerm BlankNode Blank_node_label Anon PropertyListPathNotEmpty VerbPath Path PathAlternative PathSequence PathEltOrInverse PathElt PathPrimary A_predicate Keyword PathNegatedPropertySet PathOneInPropertySet PathMod VerbSimple ObjectListPath ObjectPath GraphNodePath TriplesNodePath CollectionPath ] [ BlankNodePropertyListPath ObjectList Object GraphNode TriplesNode Collection BlankNodePropertyList PropertyListNotEmpty Verb VarOrIri PropertyListPath . GraphPatternNotTriples GroupOrUnionGraphPattern OptionalGraphPattern MinusGraphPattern GraphGraphPattern ServiceGraphPattern Filter Bind InlineData NotExistsFunc IriOrFunction DatasetClause DefaultGraphClause SourceSelector NamedGraphClause ConstructQuery ConstructTemplate ConstructTriples TriplesSameSubject PropertyList TriplesTemplate DescribeQuery AskQuery Update Update1 Load GraphRef Clear GraphRefAll Drop Add GraphOrDefault Move Copy Create InsertData QuadData Quads QuadsNotTriples DeleteData DeleteWhere QuadPattern Modify DeleteClause InsertClause UsingClause",
  maxTerm: 330,
  nodeProps: [
    ["openedBy", 213,"["],
    ["closedBy", 214,"]"]
  ],
  skippedNodes: [0,112,203,273],
  repeatNodeCount: 30,
  tokenData: "!.n~R![X^$wpq$wqr%lrs%yst.etu0avw2|wx3Xxy;syz<pz{<u{|<z|}>t}!O>y!O!P@s!P!QAu!Q![Az![!]Bi!]!^IY!^!_I_!_!`LO!`!aLT!a!bLb!b!cMi!c!}No!}#O!%}#P#Q!&z#Q#R!'P#R#S!'^#T#oNo#o#p!+y#p#q!,O#q#r!,]#y#z$w$f$g$w$g%W!%l%W%oNo%o%p!%l%p&aNo&a&b!%l&b&jNo&j1p!!i4U4d!!i4e#BY!!i#BY#BZ!,b#BZ$IS!!i$IS$I_$w$I`$Ib!!i$I|$JO$w$JT$JU$w$KV$KW$w$Kh%#t!!i&/x&Et!!i&FU&FV$w&FV;'S!!i;'S;:j!%f?&r?Ah!!i?BY?Mn!!i~$|Y'm~X^$wpq$w#y#z$w$f$g$w#BY#BZ$w$IS$I_$w$I|$JO$w$JT$JU$w$KV$KW$w&FU&FV$wR%qP'oP!_!`%tQ%yO(QQ~%|XOY&iZ]&i^r&irs)js#O&i#O#P'^#P;'S&i;'S;=`)d<%lO&i~&lXOY&iZ]&i^r&irs'Xs#O&i#O#P'^#P;'S&i;'S;=`)d<%lO&i~'^O't~~'aeOY&iZ]&i^r&irs(rsw&iwx&ix#O&i#O#P'^#P#U&i#U#V&i#V#Y&i#Y#Z&i#Z#b&i#b#c&i#c#f&i#f#g&i#g#h&i#h#i&i#i;'S&i;'S;=`)d<%lO&i~(wX't~OY&iZ]&i^r&irs'Xs#O&i#O#P'^#P;'S&i;'S;=`)d<%lO&i~)gP;=`<%l&i~)oP't~rs)r~)uVOr)rrs*[s#O)r#O#P+c#P;'S)r;'S;=`._<%lO)r~*_VOr)rrs*ts#O)r#O#P+c#P;'S)r;'S;=`._<%lO)r~*wVOr)rrs+^s#O)r#O#P+c#P;'S)r;'S;=`._<%lO)r~+cO'v~~+fcOr)rrs,qsw)rwx)rx#O)r#O#P+c#P#U)r#U#V)r#V#Y)r#Y#Z)r#Z#b)r#b#c)r#c#f)r#f#g)r#g#h)r#h#i)r#i;'S)r;'S;=`._<%lO)r~,tVOr)rrs-Zs#O)r#O#P+c#P;'S)r;'S;=`._<%lO)r~-^VOr)rrs-ss#O)r#O#P+c#P;'S)r;'S;=`._<%lO)r~-xV'v~Or)rrs+^s#O)r#O#P+c#P;'S)r;'S;=`._<%lO)r~.bP;=`<%l)r~.jV#d~OY.eYZ/PZs.est.et;'S.e;'S;=`0Z<%lO.e~/UP#d~st/X~/[VOY/qYZ/PZs/qst/Xt;'S/q;'S;=`0T<%lO/q~/tTOY/qYZ/PZ;'S/q;'S;=`0T<%lO/q~0WP;=`<%l/q~0^P;=`<%l.e~0d`!Q![1f!c!}1f#R#S1f#T#o1f%W%o1f%p&a1f&b1p1f4U4d1f4e$IS1f$I`$Ib1f$Kh%#t1f&/x&Et1f&FV;'S1f;'S;:j2v?&r?Ah1f?BY?Mn1f~1kc#n~!Q![1f!c!}1f#R#S1f#T#o1f$}%O1f%W%o1f%p&a1f&b1p1f1p4U1f4U4d1f4e$IS1f$I`$Ib1f$Je$Jg1f$Kh%#t1f&/x&Et1f&FV;'S1f;'S;:j2v?&r?Ah1f?BY?Mn1f~2yP;=`<%l1f~3PPvw3S~3XO(V~~3[XOY3wZ]3w^w3wwx6xx#O3w#O#P4l#P;'S3w;'S;=`6r<%lO3w~3zXOY3wZ]3w^w3wwx4gx#O3w#O#P4l#P;'S3w;'S;=`6r<%lO3w~4lO's~~4oeOY3wZ]3w^r3wrs3wsw3wwx6Qx#O3w#O#P4l#P#U3w#U#V3w#V#Y3w#Y#Z3w#Z#b3w#b#c3w#c#f3w#f#g3w#g#h3w#h#i3w#i;'S3w;'S;=`6r<%lO3w~6VX's~OY3wZ]3w^w3wwx4gx#O3w#O#P4l#P;'S3w;'S;=`6r<%lO3w~6uP;=`<%l3w~6}P's~wx7Q~7TVOw7Qwx7jx#O7Q#O#P8q#P;'S7Q;'S;=`;m<%lO7Q~7mVOw7Qwx8Sx#O7Q#O#P8q#P;'S7Q;'S;=`;m<%lO7Q~8VVOw7Qwx8lx#O7Q#O#P8q#P;'S7Q;'S;=`;m<%lO7Q~8qO'u~~8tcOr7Qrs7Qsw7Qwx:Px#O7Q#O#P8q#P#U7Q#U#V7Q#V#Y7Q#Y#Z7Q#Z#b7Q#b#c7Q#c#f7Q#f#g7Q#g#h7Q#h#i7Q#i;'S7Q;'S;=`;m<%lO7Q~:SVOw7Qwx:ix#O7Q#O#P8q#P;'S7Q;'S;=`;m<%lO7Q~:lVOw7Qwx;Rx#O7Q#O#P8q#P;'S7Q;'S;=`;m<%lO7Q~;WV'u~Ow7Qwx8lx#O7Q#O#P8q#P;'S7Q;'S;=`;m<%lO7Q~;pP;=`<%l7Q~;xT'n~XY<XYZ<X]^<Xpq<Xyz<k~<[TXY<XYZ<X]^<Xpq<Xyz<k~<pO$P~~<uO'p~~<zO'q~~=PQ'}~!O!P=V!Q![>V~=YP!Q![=]~=bR$x~!Q![=]!g!h=k#X#Y=k~=nR{|=w}!O=w!Q![=}~=zP!Q![=}~>SP$y~!Q![=}~>[S$w~!O!P>h!Q![>V!g!h=k#X#Y=k~>kR!Q![=]!g!h=k#X#Y=k~>yO$O~~?OQ(P~!O!P?U!Q![@U~?XP!Q![?[~?aR$|~!Q![?[!g!h?j#X#Y?j~?mR{|?v}!O?v!Q![?|~?yP!Q![?|~@RP$}~!Q![?|~@ZS${~!O!P@g!Q![@U!g!h?j#X#Y?j~@jR!Q![?[!g!h?j#X#Y?j~@xP%z~!Q![@{~AQR$t~!Q![@{!g!hAZ#X#YAZ~A^R{|Ag}!OAg!Q![Am~AjP!Q![Am~ArP$u~!Q![Am~AzO(O~~BPS$i~!O!PB]!Q![Az!g!hAZ#X#YAZ~B`R!Q![@{!g!hAZ#X#YAZ~Bnc#k~uvCy!Q![Dc![!]Dc!c!}Dc#O#PGq#R#SDc#T#oDc%W%oDc%p&aDc&b1pDc4U4dDc4e$ISDc$I`$IbDc$Kh%#tDc&/x&EtDc&FV;'SDc;'S;:jIS?&r?AhDc?BY?MnDc~C|R!Q![DV!c!iDV#T#ZDV~DYR!Q![Dc!c!iDc#T#ZDc~Dhh$`~uvCy}!ODc!O!PFS!Q![Dc![!]Dc!c!}Dc#O#PGq#R#SDc#T#oDc$}%ODc%W%oDc%p&aDc&b1pDc1p4UDc4U4dDc4e$ISDc$I`$IbDc$Je$JgDc$Kh%#tDc&/x&EtDc&FV;'SDc;'S;:jIS?&r?AhDc?BY?MnDc~FVhuvCy}!ODc!O!PFS!Q![Dc![!]Dc!c!}Dc#O#PGq#R#SDc#T#oDc$}%ODc%W%oDc%p&aDc&b1pDc1p4UDc4U4dDc4e$ISDc$I`$IbDc$Je$JgDc$Kh%#tDc&/x&EtDc&FV;'SDc;'S;:jIS?&r?AhDc?BY?MnDc~GtdqrDcstDctuDcuvDcvwDcwxDcxyDcyzDcz{Dc{|Dc|}Dc}!ODc!O!PDc!P!QDc!]!^Dc!_!`Dc!a!bDc!b!cDc#R#SDc#r#sDc~IVP;=`<%lDc~I_O#|~VIdZ(RQqrJVs!^JV!_!`KW!`!aJ{!a#OJV#P#QJV#R#SJV#T#oJV#r;'SJV;'S;=`KQ<%lOJVTJYZqrJVs!^JV!_!`JV!`!aJ{!a#OJV#P#QJV#R#SJV#T#oJV#r;'SJV;'S;=`KQ<%lOJVTKQO#iTTKTP;=`<%lJVVK]Z(TQqrJVs!^JV!_!`JV!`!aJ{!a#OJV#P#QJV#R#SJV#T#oJV#r;'SJV;'S;=`KQ<%lOJV~LTO'r~~LYP(S~!_!`L]~LbO(U~~Lg`'|~!Q![1f!c!}1f#R#S1f#T#o1f%W%o1f%p&a1f&b1p1f4U4d1f4e$IS1f$I`$Ib1f$Kh%#t1f&/x&Et1f&FV;'S1f;'S;:j2v?&r?Ah1f?BY?Mn1f~MlQ!c!}Mr#T#oMr~MwR$p~}!ONQ!c!}Mr#T#oMr~NTR!Q![N^!c!}N^#T#oN^~NcS$p~}!ONQ!Q![N^!c!}N^#T#oN^~Ntk%c~}!O!!i!O!P!$Q!Q![!!i![!]Bi!c!}No#R#SNo#T#oNo$g$}!%l$}%ONo%O%W!%l%W%oNo%o%p!%l%p&aNo&a&b!%l&b&jNo&j1p!!i1p4U!!i4U4d!!i4e$IS!!i$I`$Ib!!i$Je$Jg!!i$Kh%#t!!i&/x&Et!!i&FV;'S!!i;'S;:j!%f?&r?Ah!!i?BY?Mn!!i~!!lf}!O!!i!O!P!$Q!Q![!!i![!]Bi!c!}!!i#R#S!!i#T#o!!i$}%O!!i%W%o!!i%p&a!!i&b1p!!i1p4U!!i4U4d!!i4e$IS!!i$I`$Ib!!i$Je$Jg!!i$Kh%#t!!i&/x&Et!!i&FV;'S!!i;'S;:j!%f?&r?Ah!!i?BY?Mn!!i~!$Te}!O!!i!O!P!$Q!Q![!!i!c!}!!i#R#S!!i#T#o!!i$}%O!!i%W%o!!i%p&a!!i&b1p!!i1p4U!!i4U4d!!i4e$IS!!i$I`$Ib!!i$Je$Jg!!i$Kh%#t!!i&/x&Et!!i&FV;'S!!i;'S;:j!%f?&r?Ah!!i?BY?Mn!!i~!%iP;=`<%l!!i~!%qS%c~!c!}!%l#R#S!%l#T#o!%l$g&j!%l~!&ST%n~XY!&cYZ!&c]^!&cpq!&c#P#Q!&u~!&fTXY!&cYZ!&c]^!&cpq!&c#P#Q!&u~!&zO%X~~!'PO%m~~!'UP'z~#Q#R!'X~!'^O$q~~!'cT%c~![!]!'r!c!}!%l#R#S!%l#T#o!%l$g&j!%l~!'u`!Q![!(w!c!}!(w#R#S!(w#T#o!(w%W%o!(w%p&a!(w&b1p!(w4U4d!(w4e$IS!(w$I`$Ib!(w$Kh%#t!(w&/x&Et!(w&FV;'S!(w;'S;:j!+s?&r?Ah!(w?BY?Mn!(w~!(|e%W~}!O!(w!O!P!*_!Q![!(w!c!}!(w#R#S!(w#T#o!(w$}%O!(w%W%o!(w%p&a!(w&b1p!(w1p4U!(w4U4d!(w4e$IS!(w$I`$Ib!(w$Je$Jg!(w$Kh%#t!(w&/x&Et!(w&FV;'S!(w;'S;:j!+s?&r?Ah!(w?BY?Mn!(w~!*be}!O!(w!O!P!*_!Q![!(w!c!}!(w#R#S!(w#T#o!(w$}%O!(w%W%o!(w%p&a!(w&b1p!(w1p4U!(w4U4d!(w4e$IS!(w$I`$Ib!(w$Je$Jg!(w$Kh%#t!(w&/x&Et!(w&FV;'S!(w;'S;:j!+s?&r?Ah!(w?BY?Mn!(w~!+vP;=`<%l!(w~!,OO'x~V!,TP'{S#p#q!,WR!,]O(WR~!,bO'w~~!,gq'm~X^$wpq$w}!O!!i!O!P!$Q!Q![!!i![!]Bi!c!}!!i#R#S!!i#T#o!!i#y#z$w$f$g$w$}%O!!i%W%o!!i%p&a!!i&b1p!!i1p4U!!i4U4d!!i4e#BY!!i#BY#BZ!,b#BZ$IS!!i$IS$I_$w$I`$Ib!!i$I|$JO$w$JT$JU$w$Je$Jg!!i$KV$KW$w$Kh%#t!!i&/x&Et!!i&FU&FV$w&FV;'S!!i;'S;:j!%f?&r?Ah!!i?BY?Mn!!i",
  tokenizers: [0, 1, 2],
  topRules: {"Unit":[0,113]},
  specialized: [{term: 203, get: (value, stack) => (caseInsensitive(value) << 1), external: caseInsensitive},{term: 203, get: (value) => spec_Keyword[value] || -1}],
  tokenPrec: 5071
});

const SparqlLanguage = LRLanguage.define({
    parser: parser.configure({
        props: [
            styleTags({
                Comment: tags.comment,
                // Query type and clause keywords
                "BASE PREFIX SELECT CONSTRUCT DESCRIBE ASK WHERE DISTINCT REDUCED FROM NAMED": tags.keyword,
                // Graph pattern keywords
                "UNION OPTIONAL MINUS GRAPH SERVICE SILENT FILTER BIND AS VALUES UNDEF": tags.keyword,
                // Solution modifier keywords
                "GROUP BY HAVING ORDER ASC DESC LIMIT OFFSET": tags.keyword,
                // Update operation keywords
                "LOAD INTO TO CLEAR DROP CREATE ADD MOVE COPY INSERT DATA DELETE WITH DEFAULT ALL USING": tags.keyword,
                // Built-in function keywords
                "STR LANG LANGMATCHES DATATYPE BOUND IRI URI BNODE RAND ABS CEIL FLOOR ROUND": tags.keyword,
                "CONCAT STRLEN UCASE LCASE ENCODE_FOR_URI CONTAINS STRSTARTS STRENDS STRBEFORE STRAFTER": tags.keyword,
                "YEAR MONTH DAY HOURS MINUTES SECONDS TIMEZONE TZ NOW UUID STRUUID": tags.keyword,
                "MD5 SHA1 SHA256 SHA384 SHA512 COALESCE IF STRLANG STRDT SAMETERM": tags.keyword,
                "ISIRI ISURI ISBLANK ISLITERAL ISNUMERIC EXISTS NOT IN SUBSTR REPLACE REGEX": tags.keyword,
                // Aggregate function keywords
                "COUNT SUM MIN MAX AVG SAMPLE GROUP_CONCAT SEPARATOR": tags.keyword,
                "Var ObjectListPath/..": tags.variableName,
                String: tags.string,
                Integer: tags.integer,
                "Double Decimal": tags.float,
                "{ }": tags.brace,
                Langstag: tags.annotation,
                "TRUE FALSE": tags.bool,
                "VerbPath Namespace IriRef": tags.namespace,
                "Iri": tags.url,
                "RDFLiteral/Iri": tags.typeName,
            }),
            foldNodeProp.add({
                GroupGraphPattern: foldInside,
                QuadData: foldInside,
                // +7 skips the leading keyword ("PREFIX " or "BASE   ") so the fold starts after it
                Prologue(tree) { return { from: tree.from + 7, to: tree.to }; }
            }),
        ]
    }),
});
function sparql() {
    return new LanguageSupport(SparqlLanguage);
}

export { SparqlLanguage, sparql };
